src/cache.ts

```diff
@@ -0,0 +1,114 @@
+import { KubernetesObject } from './types';
+import { Watch } from './watch';
+
+export interface ObjectCache<T> {
+    get(name: string, namespace?: string): T | null;
+    list(namespace?: string): T[];
+}
+
+export type ListCallback<T extends KubernetesObject> = (list: T[]) => void;
+
+export class ListWatch<T extends KubernetesObject> implements ObjectCache<T> {
+    private objects: T[] = [];
+    private indexCache: any = {};
```

Comment:
I think you might be able to pin the type down to `{ [key: string]: T[] }` which would give you additional checking.

Reply:
Done (learning more Typescript by the day!)